File: server/internal/transport/dto/types/overview.go
package dto

type AgentsOverview struct {
	Summary *AgentsSummary `json:"summary"`
	TopN    *TopNAgents    `json:"topN"`
} //@name AgentsOverview

type AgentsSummary struct {
	TotalAgents             int               `json:"totalAgents"`
	OnlineAgents            int               `json:"onlineAgents"`
	AverageCPUUsage         float64           `json:"averageCPUUsage"`
	AverageMemoryUsage      float64           `json:"averageMemoryUsage"`
	CPUUsageDistribution    CountDistribution `json:"cpuUsageDistribution"`
	MemoryUsageDistribution CountDistribution `json:"memoryUsageDistribution"`
	DiskUsageDistribution   CountDistribution `json:"diskUsageDistribution"`
} //@name AgentsSummary

type CountDistribution struct {
	High   CountByPercent `json:"high"`
	Medium CountByPercent `json:"medium"`
	Low    CountByPercent `json:"low"`
} //@name CountDistribution

type CountByPercent struct {
	Percent float64 `json:"percent"`
	Count   int     `json:"count"`
} //@name CountByPercent

type AgentWithCPUUsageDTO struct {
	AgentShortDTO
	CPUUsage float64 `json:"cpuUsage"`
} //@name AgentWithCPUUsageDTO

type AgentWithMemoryUsageDTO struct {
	AgentShortDTO
	MemoryUsage float64 `json:"memoryUsage"`
} //@name AgentWithMemoryUsageDTO
type TopNAgents struct {
	CPUUsage    []AgentWithCPUUsageDTO    `json:"cpuUsage"`
	MemoryUsage []AgentWithMemoryUsageDTO `json:"memoryUsage"`
} //@name TopNAgents

File: web-panel/src/domain/overview.ts


export interface CountDistribution {
    high: CountByPercent;
    medium: CountByPercent;
    low: CountByPercent;
}

export const CountDistToPieList = (dist: CountDistribution) => {
    return [
        {
            Name: `>${dist.high.percent}`,
            Value: dist.high.count,
            fill: "#c94b4b",
        },
        {
            Name: `>${dist.medium.percent}`,
            Value: dist.medium.count,
            fill: "#c9ac4b",
        },
        {
            Name: `>${dist.low.percent}`,
            Value: dist.low.count,
            fill: "#4bc95c",
        }
    ]
}

export interface CountByPercent {
    percent: number;
    count: number;
}

export interface AgentsSummary {
    averageCPUUsage: number;
    averageMemoryUsage: number;
    cpuUsageDistribution: CountDistribution;
    memoryUsageDistribution: CountDistribution;
    diskUsageDistribution: CountDistribution;
    onlineAgents: number;
    totalAgents: number;
}

export interface TopNAgents {
    cpuUsage: AgentWithCPUUsage[];
    memoryUsage: AgentWithMemoryUsage[];
}

export interface AgentWithCPUUsage {
    id: string;
    name: string;
    cpuUsage: number;
}

export interface AgentWithMemoryUsage {
    id: string;
    name: string;
    memoryUsage: number;
}

export interface Overview {
    summary: AgentsSummary;
    topN: TopNAgents;
}